### ISSUE TITLE
Add a note to versions.md about ClusterClass issues on Kubernetes v1.23.{0,1,2}

### ISSUE THREAD

**Issue by sbueringer** [kubernetes-sigs/cluster-api#5990]
There was an announcement that there is an issue with Kubernetes v1.23.{0,1,2} and CRDs using `x-kubernetes-preserve-unknown-fields: true` (https://groups.google.com/a/kubernetes.io/g/dev/c/Xl1sm-CItaY https://github.com/kubernetes/kubernetes/issues/107690)

ClusterAPI uses preserve unknown fields only for ClusterClass patches and variables:
* ClusterClass: jsonPatches[x].value, openAPIV3Schema.{default,enum,example}
* Cluster: variables[x].value, variables.overrides[x].value

A concrete example. When the following YAML is deployed:
```yaml
  - name: nestedArrayVar
    required: true
    schema:
      openAPIV3Schema:
        type: array
        items:
          type: object
          properties:
            nestedField:
              type: string
        default:
        - nestedField: "abc"
        - nestedField: "def"
```
the apiserver will prune nested fields in the default value:
```yaml
  - name: nestedArrayVar
    required: true
    schema:
      openAPIV3Schema:
        default:
        - {}
        - {}
        items:
          properties:
            nestedField:
              type: string
          type: object
        type: array
```
This can happen with all other fields mentioned above. 

Because of this issue it's not recommended to use ClusterClass patches and variables on a management cluster with Kubernetes v1.23.{0,1,2}.

Let's add a **short** note to versions.md (https://cluster-api.sigs.k8s.io/reference/versions.html) that it's not recommened to use ClusterClass with patches on those Kubernetes versions and link back to the current issue.

/kind feature

**Comment by liggitt** [comment]
FYI, v1.23.3 has been released with a fix for kubernetes/kubernetes#107690

**Comment by sbueringer** [comment]
> FYI, v1.23.3 has been released with a fix for [kubernetes/kubernetes#107690](https://github.com/kubernetes/kubernetes/issues/107690)

Thx!

For completeness, simple arrays like e.g. a string array are not affected.

**Comment by sbueringer** [comment]
I also opened an issue to switch to 1.23.3 in our quickstart doc and in CI: https://github.com/kubernetes-sigs/cluster-api/issues/5995
